src/components/include/utils/threads/thread.h

```diff
@@ -114,6 +114,7 @@ class Thread {
   // Should be locked to protect isThreadRunning_ and thread_created_ values
   sync_primitives::Lock state_lock_;
   volatile bool isThreadRunning_;
+  volatile bool do_iteration_;
```

Comment:
@orhan-mehmedov I think this point (I mean enum providing) you could implement within this task. But please confirm with your TL your timelines.